tsdb/head_read.go

```diff
@@ -86,6 +86,15 @@ func (h *headIndexReader) LabelValues(ctx context.Context, name string, matchers
 	return labelValuesWithMatchers(ctx, h, name, matchers...)
 }
 
+// LabelValuesFiltered returns values for which the filter function returns true.
+func (h *headIndexReader) LabelValuesFiltered(ctx context.Context, name string, filter func(string) bool) ([]string, error) {
+	if h.maxt < h.head.MinTime() || h.mint > h.head.MaxTime() {
+		return []string{}, nil
```

Comment:
Nit: is there a reason why you don't just return `nil`? That's idiomatic go, right?

```suggestion
		return nil, nil
```

Reply:
Just copy-paste from `LabelValues` on line 79.

Reply:
I did as you suggested; note there were a couple of tests where this made a difference.